src/heartratetask/HeartRateTask.cpp

```diff
@@ -65,8 +65,17 @@ void HeartRateTask::Work() {
     }
 
     if (measurementStarted) {
-      ppg.Preprocess(static_cast<float>(heartRateSensor.ReadHrs()));
-      auto bpm = ppg.HeartRate();
+      auto start = xTaskGetTickCount();
+      auto ambient = ppg.Preprocess(static_cast<float>(heartRateSensor.ReadHrs()), heartRateSensor.ReadAls());
+      int bpm = ppg.HeartRate();
+      auto end = xTaskGetTickCount();
+      total = end - start;
```

Comment:
These don't seem to be used.

Reply:
This was used to time the algorithm execution time in milliseconds. Removed.